scripts/populate_ingest/populate_ingest_from_nxn.py

```diff
@@ -0,0 +1,178 @@
+# --- core imports
+import argparse
+from datetime import datetime
+import itertools
+import logging
+import pprint
+import re
+import sys
+
+# --- third party library imports
+import Levenshtein
+
+# --- application imports
+from convert.conversion_utils import get_accessions
+from convert.europe_pmc import EuropePmcConverter
+from data_entities.nxn_db import NxnDatabase
+from services.europe_pmc import EuropePmc
+from services.nxn_db import NxnDatabaseService
+from services.ingest import QuickIngest
+
+
+# INGEST_URL = "https://api.ingest.archive.data.humancellatlas.org"
+INGEST_URL = "http://localhost:8080"
+
+# utility functions
+def reformat_title(title: str) -> str:
+    return re.sub("\W", "", title).lower().strip()
+
+def get_distance_metric(title1: str,title2: str):
+    if not all([title1, title2]):
+        return 0
+    max_len = max(len(title1), len(title2))
+    dist_metric = 100-(Levenshtein.distance(title1, title2)/max_len)*100
+    return dist_metric
+
+def get_file_content(file_path):
+    with open(file_path, "r") as file_path:
+        return file_path.read()
+
+def prepare_logging():
+    logging.basicConfig(filename='nxn_db.log', format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
```

Comment:
Ah, it's nice you're saving logs into a file.  I usually just create a report file in the end.  I think it's a good practice for ad-hoc/ops scripts like this.

Reply:
Logging could involve multiple locations in different granularities.
For example: high level events (start, stop, etc.) to the console, while deeper debug info goes in the log file.
see, for example: https://docs.python.org/3/howto/logging-cookbook.html#logging-to-multiple-destinations

Reply:
@amnonkhen  I followed the cookbook above and that works, but now all exceptions get logged to the console as well.

I'd added the exception related code in this function: https://github.com/ebi-ait/hca-ebi-dev-team/blob/d3ad3f77f05710c3a35001ed313732891e686afb/scripts/populate_ingest/populate_ingest_from_nxn.py#L42 (following this stackoverflow answer: https://stackoverflow.com/questions/6234405/logging-uncaught-exceptions-in-python)
to make sure exceptions only get logged to the log file, but that's not working after the following the cookbook.

Would you have any ideas / suggestions?